src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -2937,6 +2996,50 @@ internal static string ReplaceAlias(this IEdmModel model, string name)
             return name;
         }
 
+        #region IEdmTargetPath extensions
+
+        /// <summary>
+        /// Get the path string from the <see cref="IEdmTargetPath"/>.
+        /// </summary>
+        /// <param name="targetPath">The target path.</param>
+        /// <returns>The target path string.</returns>
+        public static string GetPathString(this IEdmTargetPath targetPath)
+        {
+            EdmUtil.CheckArgumentNull(targetPath, nameof(targetPath));
+
+            List<IEdmElement> segments = targetPath.Segments.ToList();
+            IEdmEntityContainer container = segments[0] as IEdmEntityContainer;
+            IEdmEntityContainerElement containerElement = segments[1] as IEdmEntityContainerElement;
+
+            if (container == null || containerElement == null)
+            {
+                return null;
+            }
+

```

Comment:
Be noted: We don't need to test those. If the input "targetPath" is not null, it means this object was constructed correctly, that is, in the constructor, we finished the verification. Otherwise, you can't get a valid targetPath, right?

Reply:
That's correct.
Unless the customer has their custom implementation of the `IEdmTargetPath` and they don't enforce the validations we have in the `EdmTargetPath`